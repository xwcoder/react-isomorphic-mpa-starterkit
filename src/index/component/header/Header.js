import React from 'react'

export default function Header ({ title }) {
  return (
    <h1 className="app-header">{title}</h1>
  )
}
