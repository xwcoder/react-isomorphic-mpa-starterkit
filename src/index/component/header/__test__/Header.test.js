import React from 'react'
import { shallow } from 'enzyme'
import Header from '../Header'

describe('index/component/Header', () => {

  it('header with title', () => {
    const wrapper = shallow(<Header title="今日要闻" />)
    expect(wrapper.find('h1')).toHaveLength(1)
    expect(wrapper.hasClass('app-header')).toBeTruthy()
    expect(wrapper.text()).toEqual('今日要闻')
    expect(wrapper).toMatchSnapshot()
  })

  it('header with empty title', () => {

    const wrapper = shallow(<Header />)
    expect(wrapper.find('h1')).toHaveLength(1)
    expect(wrapper.hasClass('app-header')).toBeTruthy()
    expect(wrapper.text()).toEqual('')
    expect(wrapper).toMatchSnapshot()
  })
})
