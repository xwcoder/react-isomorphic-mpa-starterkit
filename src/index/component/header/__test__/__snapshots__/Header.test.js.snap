// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index/component/Header header with empty title 1`] = `
<h1
  className="app-header"
/>
`;

exports[`index/component/Header header with title 1`] = `
<h1
  className="app-header"
>
  今日要闻
</h1>
`;
